TheJumpCloud/support
# Sa 3067 get jc system app

## Issues
* [SA-3067](https://jumpcloud.atlassian.net/browse/sa-3067) - SA-3067-Get-JCSystemApp

## What does this solve?
Enables admins to search software installed in their systems
## Is there anything particularly tricky?
Not really, if `-search `is not used filters are case sensitive especially `-softwareName`. 
-search can be time consuming since it searches all the systems in the org
## How should this be tested?
Run:
`Get-JCSystemApp` - returns all software in all systems
`Get-JCSystemApp -systemId "YOURID"` - returns all software installed in the specific system
`Get-JCSystemApp -systemOs "Windows"` - also test MacOs and Linux, this should return all apps in specific OS
`Get-JCSystemApp -systemOs "Windows" -softwareName "Microsoft Edge"`
`Get-JCSystemApp -systemOs "Windows" -softwareName "Microsoft Edge" -version "108.0.1462.42"`
Repeat the last 3 but instead of `-systemOs` use `-systemId`
`Get-JCSystemApp -softwareName "disk utility" -search`
`Get-JCSystemApp -softwareName "disk utility"  -systemId -search`


## Screenshots

![image](https://user-images.githubusercontent.com/97972790/214908039-bb3dda08-06e8-40e3-8de7-a940b45918c0.png)
![image](https://user-images.githubusercontent.com/97972790/214908203-b8b5ff74-c2d7-4cf1-acb8-5382629a51a6.png)



[SA-3067]: https://jumpcloud.atlassian.net/browse/SA-3067?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
